build.xml

```diff
@@ -528,7 +528,9 @@
           </dependency>
           <dependency groupId="org.mockito" artifactId="mockito-core" version="3.2.4" scope="test"/>
           <dependency groupId="org.quicktheories" artifactId="quicktheories" version="0.26" scope="test"/>
-          <dependency groupId="com.google.code.java-allocation-instrumenter" artifactId="java-allocation-instrumenter" version="${allocation-instrumenter.version}" scope="test"/>
+          <dependency groupId="com.google.code.java-allocation-instrumenter" artifactId="java-allocation-instrumenter" version="${allocation-instrumenter.version}" scope="test">
+            <exclusion groupId="com.google.guava" artifactId="guava"/>
+          </dependency>
```

Comment:
interesting, `test-memory` target is was broken already.

it throws `Error opening zip file or JAR manifest missing : build/lib/jars/java-allocation-instrumenter-3.1.0.jar`
as that jar file is now test scope and only found under build/test/lib/jar/ now
ref: https://github.com/apache/cassandra/blob/cassandra-4.0/build.xml#L1609

@jacek-lewandowski would you mind if we fix that in this PR?  
As I cannot see the use of `import com.google.monitoring.runtime.instrumentation` anywhere outside of tests, 
so line 1609 should be instead
```
<jvmarg value="-javaagent:${build.dir}/test/lib/jars/java-allocation-instrumenter-${allocation-instrumenter.version}.jar"/>
```